backend/assignments/2/src/main/java/com/kdu/caching/exceptions/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,53 @@
+package com.kdu.caching.exceptions;
+
+import com.kdu.caching.dto.ErrorDTO;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+/**
+ * Global exception handler for handling exceptions across the application.
+ */
+@ControllerAdvice
+public class GlobalExceptionHandler {
```

Comment:
Will all exceptions be of type `BAD_REQUEST`. What if request was valid but something failed internally?